### PR TITLE
Fix relocated readonly data in custom sections

### PR DESCRIPTION
**Pull request by bjorn3**
Lld doesn't allow relocations in readonly sections

This is required for https://github.com/bjorn3/rustc_codegen_cranelift/issues/1055.

### CODE REVIEW

**Review by pchickey — APPROVED** [review]

**Comment by bjorn3** [comment]
@pckickey Are you waiting on something before merging? CI passed.

**Comment by pchickey** [comment]
Ah sorry I just forgot. Thanks for the reminder :)

**Pull request merged**
